### PR TITLE
Implement DataFrame.info

### LINKED ISSUE #872: DataFrame.info
Implement `DataFrame.info`.

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.info.html
`memory usage` is not needed for Koalas.

### PR DESCRIPTION
**Pull request by HyukjinKwon**
This PR proposes `DataFrame.info`.

```python
>>> import databricks.koalas as ks
>>> ks.range(100).info()
<class 'databricks.koalas.frame.DataFrame'>
Index: 100 entries, 0 to 99
Data columns (total 1 columns):
id    100 non-null int64
```

Resolves #872

### CODE REVIEW

File: databricks/koalas/frame.py [hunk 1]
```diff
@@ -8036,6 +8036,100 @@ def idxmin(self, axis=0):
...
+        dtypes: float64(1), int64(1), object(1)
+
+        Pipe output of DataFrame.info to buffer instead of sys.stdout, get
+        buffer content and writes to a text file:
+
+        >>> import io
+        >>> buffer = io.StringIO()
+        >>> df.info(buf=buffer)
+        >>> s = buffer.getvalue()
+        >>> path = "/tmp/test"
```
Comment (ueshin) [inline databricks/koalas/frame.py]: We have a fixture named `path`?

File: databricks/koalas/frame.py [hunk 2]
```diff
@@ -8036,6 +8036,100 @@ def idxmin(self, axis=0):
...
+        >>> with open('%s/info/foo.txt' % path, "w",
+        ...           encoding="utf-8") as f:
+        ...     _ = f.write(s)
+        """
+        # To avoid pandas' existing config affects Koalas.
+        # TODO: should we have corresponding Koalas configs?
```
Comment (ueshin) [inline databricks/koalas/frame.py]: Yeah, we should in a separate PR.

File: databricks/koalas/frame.py [hunk 3]
```diff
@@ -8036,6 +8036,99 @@ def idxmin(self, axis=0):
...
+        with pd.option_context(
+                'display.max_info_columns', sys.maxsize,
+                'display.max_info_rows', sys.maxsize):
+            try:
+                self._data = self  # hack to use pandas' info as is.
+                return pd.DataFrame.info(
+                    self, verbose=verbose, buf=buf, max_cols=max_cols,
+                    memory_usage=False, null_counts=null_counts)
```
Comment (ueshin) [inline databricks/koalas/frame.py]: I'm wondering how many Spark jobs are kicked.

Comment (HyukjinKwon) [reply databricks/koalas/frame.py]: It seems launches 4 jobs. `_summary` twice (for index, and columns) and `len` twice. `len` can be saved once but I guess it's fine for now.

Comment (ueshin) [reply databricks/koalas/frame.py]: Does it depend on the column number?

Comment (HyukjinKwon) [reply databricks/koalas/frame.py]: I just checked it too. Nope, it doesn't.

Comment (ueshin) [reply databricks/koalas/frame.py]: ah, sounds great.

**Pull request merged**